internal/variant/variant.go

```diff
@@ -34,17 +34,68 @@ package variant
 import (
 	"encoding/asn1"
 	"fmt"
+	"sort"
+
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
 )
 
 const (
 	dummy              = "dummy"
 	awsNitroTPM        = "aws-nitro-tpm"
 	gcpSEVES           = "gcp-sev-es"
 	azureSEVSNP        = "azure-sev-snp"
-	azureTrustedLaunch = "azure-trustedlaunch"
+	azureTrustedLaunch = "azure-trusted-launch"
```

Comment:
I dont think we want to change the string representation of this value, since this could break compatibility with older images.
Maybe @malt3 has an opinion on this?